### DISCUSSION TITLE
Where to add information about the shipping contractor in a TradeLineItem?

### DISCUSSION THREAD

**Discussion by AlexanderPilhar** [stephanstapel/ZUGFeRD-csharp#222] [Q&A]
Hi everyone!

I would like to know how where to add information about the shipping contractor in a `TradLineItem`?

The specification version is 2.1. The profile will have to be "Extended", I guess!?

I appriciate the help!

**EDIT**

I have an Invoice with a list of `TradeLineItems` that looks like this:

| Delivery Note | Date | Material | Recipient | Shippment Carrier / License Plate | Quantity | Price per Unit | Sum Total |
| --------------- | ----- | --------- | ---------- | ------------------------------------- | --------- | --------------- | ----------- | 
| 1 | 01.11.2023 | Mat_01 | Recipient_01 | Carrier_01 / XXX_01 | 10 | € 100,00 | €1.000,00 |
| 2 | 01.11.2023 | Mat_02 | Recipient_01 | Carrier_01 / XXX_01 | 10 | € 200,00 | €2.000,00 |
| 2 | 01.11.2023 | Mat_01 | Recipient_02 | Carrier_02 / XXX_02 | 10 | € 100,00 | €1.000,00 |

So, each `TradeLineItem` can have an **individual recipient** and also an **indvidual carrier**.

Additional Information:
The invoice is created by the producer and sent to the customer. The customer himself has multiple customers (recipients) to where the material is shipped to. Each shippment is done with an individual carrier. The material is shipped from the producer to the recipient.

Note: I don't know if this is possible to do with ZUGFeRD/Factur-X at all. I seem to be unable to find anything useful about it in the specification.

**Comment by stephanstapel** [comment]
Could yo elaborate which entity you are refering to? I don't understand 'shipping contractor'/ can't map this to ZUGFeRD entities.

**Reply by AlexanderPilhar** [reply]
Sorry for being too vague about what I want to achieve - I will update the question with more information!
